framework.py

```diff
@@ -9,32 +9,169 @@
 # not expressly granted therein are reserved by Shotgun Software Inc.
 
 import sgtk
+import sys
 import os
+import struct
+from sgtk.util import LocalFileStorageManager
+
 
 class DesktopserverFramework(sgtk.platform.Framework):
+    """
+    Provides browser integration.
+    """
+
+    def __init__(self, *args, **kwargs):
+        super(DesktopserverFramework, self).__init__(*args, **kwargs)
+        self._server = None
+        self._settings = None
+        self._tk_framework_desktopserver = None
 
     ##########################################################################################
     # init and destroy
+    def launch_desktop_server(self):
+        """
+        Initializes the desktop server.
+        """
+        self._tk_framework_desktopserver = self.import_module("tk_framework_desktopserver")
+
+        # Read the browser integration settings from disk. By passing in location=None, the Toolkit API will be
+        # used to locate the settings instead of looking at a specific file.
+        self._settings = self._tk_framework_desktopserver.Settings(
+            location=None,
+            default_certificate_folder=os.path.join(
+                LocalFileStorageManager.get_global_root(
+                    LocalFileStorageManager.CACHE, LocalFileStorageManager.CORE_V18
+                ),
+                "desktop",
+                "config",
+                "certificates"
+            )
+        )
+        self._settings.dump(self.logger)
+
+        # Twisted only runs on 64-bits.
+        if not self.__is_64bit_python():
+            self.logger.warning("The browser integration is only available with 64-bit versions of Python.")
+            self._integration_enabled = False
+        # Did the user disable it?
+        elif self._settings.integration_enabled:
+            self.logger.info("Browser integration has been disabled in the Toolkit settings.")
+            self._integration_enabled = False
+        else:
+            self._integration_enabled = True
+
+        if not self._integration_enabled:
+            return
 
-    def init_framework(self):
-        self.log_debug("%s: Initializing..." % self)
-        self.server = None
+        try:
+            self.__ensure_certificate_ready()
+
+            self._server = self._tk_framework_desktopserver.Server(
+                port=self._settings.port,
+                low_level_debug=self._settings.low_level_debug,
+                whitelist=self._settings.whitelist,
+                keys_path=self._settings.certificate_folder
+            )
+
+            self._server.start()
+        except:
```

Comment:
I'd prefer an `except Exception:` here. It'll catch anything we care about, and will allow things like SystemExit and KeyboardInterrupt to still bubble up. It's also strongly suggested by pep8.